storage/coin_storage_test.go

```diff
@@ -373,6 +397,45 @@ func TestCoinStorage(t *testing.T) {
 		assert.Equal(t, &types.CoinIdentifier{Identifier: "coin4"}, coinIdentifier)
 		assert.Equal(t, blockIdentifier, block)
 	})
+
+	t.Run("AddCoins", func(t *testing.T) {
+		accountCoins := []*AccountCoin{
```

Comment:
Can we try adding a coin that already exists to make sure we don't put it in twice?